client/src/app/routes/stardust/Foundry.tsx

```diff
@@ -45,54 +51,33 @@ const Foundry: React.FC<RouteComponentProps<FoundryProps>> = (
                 output.unlockConditions[0] as IImmutableAliasUnlockCondition;
             const aliasId = (immutableAliasUnlockCondition.address as IAliasAddress).aliasId;
 
+            const immutableFeatures = (foundryDetails?.output as IFoundryOutput).immutableFeatures;
+            const metadataFeature = immutableFeatures?.find(
+                feature => feature.type === METADATA_FEATURE_TYPE
+            ) as IMetadataFeature;
+            const parsedMetadata = tryParseMetadata<ITokenMetadata>(metadataFeature.data, tokenSchemeIRC30);
```

Comment:
If `metadataFeature` is not found the code can explode here on `.data`. 
Reproduced with this one `testnet/foundry/0x0867478068f5e6efe55b9d2ef01aa02eccbbdd1a25f68694f71b6e6423206ac8d90100000000`